components/FaceScanner/FaceScanner.tsx

```diff
@@ -0,0 +1,215 @@
+import {selectIsLivenessEnabled} from '../../machines/settings';
+import {GlobalContext} from '../../shared/GlobalContext';
+import {selectIsActive} from '../../machines/app';
+import {Theme} from '.././ui/styleUtils';
+import {getRandomInt} from '../../shared/commonUtil';
+import {
+  checkBlink,
+  cropEyeAreaFromFace,
+  faceDetectorConfig,
+  getFaceBounds,
+  imageCaptureConfig,
+} from './FaceScannerHelper';
+import LivenessDetection from './LivenessDetection';
+import FaceCompare from './FaceCompare';
+
+export const FaceScanner: React.FC<FaceScannerProps> = props => {
+  const {t} = useTranslation('FaceScanner');
+  const {appService} = useContext(GlobalContext);
+  const settingsService = appService?.children?.get('settings') || {};
+  const isActive = useSelector(appService, selectIsActive);
+
+  const machine = useRef(createFaceScannerMachine(props.vcImage));
+  const service = useInterpret(machine.current);
+
+  const whichCamera = useSelector(service, selectWhichCamera);
+  const cameraRef = useSelector(service, selectCameraRef);
+  const livenessEnabled = useSelector(settingsService, selectIsLivenessEnabled);
+
+  const isPermissionDenied = useSelector(service, selectIsPermissionDenied);
+  const isValid = useSelector(service, selectIsValid);
+  const isInvalid = useSelector(service, selectIsInvalid);
+  const isCheckingPermission = useSelector(service, selectIsCheckingPermission);
+  const isScanning = useSelector(service, selectIsScanning);
+  const isCapturing = useSelector(service, selectIsCapturing);
+  const isVerifying = useSelector(service, selectIsVerifying);
+
+  const [counter, setCounter] = useState(0);
+  const [screenColor, setScreenColor] = useState('#0000ff');
+  const [faceToCompare, setFaceToCompare] = useState(null);
+  const [opacity, setOpacity] = useState(1);
+  const [picArray, setPicArray] = useState([]);
+
+  const screenFlashColors = ['#0000FF', '#00FF00', '#FF0000'];
+  const MAX_COUNTER = 15;
+
+  const randomNumToFaceCompare = getRandomInt(counter, MAX_COUNTER - 1);
+  const [infoText, setInfoText] = useState<string>(t('livenessCaptureGuide'));
+
+  const setCameraRef = useCallback(
+    (node: Camera) => {
+      if (node != null && !isScanning) {
+        service.send(FaceScannerEvents.READY(node));
+      }
+    },
+    [isScanning],
+  );
+
+  function handleOnCancel() {
+    props.onCancel();
+  }
+
+  async function captureImage(screenColor) {
+    try {
+      if (cameraRef) {
+        const capturedImage = await cameraRef.takePictureAsync(
+          imageCaptureConfig,
+        );
+
+        setPicArray([...picArray, {color: screenColor, image: capturedImage}]);
+
+        if (counter === randomNumToFaceCompare) {
+          setFaceToCompare(capturedImage);
+        }
+      }
+    } catch (error) {
+      console.error('Error capturing image:', error);
+    }
+  }
+
+  async function handleFacesDetected({faces}) {
+    checkBlink(faces[0]);
+
+    if (!livenessEnabled) {
```

Comment:
Do we need this check as this function is being used only in LivenessDetection component?

If it's needed, then shouldn't be it the first line instead of checking blink first?

Reply:
Yes removed the if block, it is not needed now.